src/WebJobs.Extensions/Extensions/Timers/Bindings/TimerTriggerBinding.cs

```diff
@@ -66,7 +66,7 @@ public async Task<ITriggerData> BindAsync(object value, ValueBindingContext cont
                 ScheduleStatus status = null;
                 if (_attribute.UseMonitor && _scheduleMonitor != null)
                 {
-                    status = await _scheduleMonitor.GetStatusAsync(_timerName);
+                    status = await _scheduleMonitor.GetSafeStatusAsync(_timerName);
```

Comment:
The bug was here -- I wasn't correcting any invalid dates as I was in TimerListener.